src/saltext/vmware/modules/storage_policies.py

```diff
@@ -0,0 +1,477 @@
+                                dynamicType = <unset>,
+                                dynamicProperty = (vmodl.DynamicProperty) [],
+                                propertyInstance = (pbm.capability.PropertyInstance) [
+                                (pbm.capability.PropertyInstance) {
+                                    dynamicType = <unset>,
+                                    dynamicProperty = (vmodl.DynamicProperty) [],
+                                    id = 'proportionalCapacity',
+                                    operator = <unset>,
+                                    value = 0
+                                }
+                                ]
+                            }
+                        ]
+                    }
+                    ],
+                    forceProvision = <unset>
+                }
+            ]
+        },
+        generationId = 0,
+        isDefault = false,
+        systemCreatedProfileType = <unset>,
+        lineOfService = <unset>
+        }
+
+    Args:
+        pbmSi: PBM Service Instance
+
+    Returns:
+        list: of profile VMOMI objects
+    """
+    resourceType = pbm.profile.ResourceType(
+        resourceType=pbm.profile.ResourceTypeEnum.STORAGE
+    )
+
+    profileManager = pbmSi.RetrieveContent().profileManager
+    profileIds = profileManager.PbmQueryProfile(resourceType)
+    profiles = profileManager.PbmRetrieveContent(profileIds)
+    return profiles
+
+def find(policy_name=None, service_instance=None, profile=None):
+    """
+    Gets storage policies. Returns list.
+
+    policy_name
+        Filter by policy name, if None returns all policies
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    profile
+        Profile to use (optional)
+    """
+    service_instance = service_instance or connect.get_service_instance(
+        config=__opts__, profile=profile
+    )
+    
+    pbmSi = _connect_spbm_(service_instance._stub, service_instance._stub.schemeArgs['context'])
+    policies = _get_vsan_storage_policies_(pbmSi)
+    
+    result_policies = []
+    for policy in policies:
+        if isinstance(policy, pbm.profile.CapabilityBasedProfile):
+            if policy_name is None or policy.name == policy_name:
+                result_policies.append(policy)
+    
+    # make JSON representation of current policies
+    # old_configs holds only the rules that are in the scope of interest (provided in argument config_input)
+    result = []
+    for policy in result_policies:
+        policy_json = {
+            "policyName": policy.name
+        }
+        policy_json['constraints'] = []
+        try:
+            for sub_profile in policy.constraints.subProfiles:
+                policy_constrain = {
+                    "name": sub_profile.name
+                }
+                policy_json['constraints'].append(policy_constrain)
```

Comment:
**suggestion (non-blocking)**: it's a bit of a balancing act when to create a variable vs just use a constant. This *could* be:

```suggestion
                policy_json["constraints"].append({"name": sub_profile.name})
````

Typically I'll use a variable if it makes future debugging attempts easier. In this case, there shouldn't be much of a difference, so _I_ would just use the inline approach.